common/djangoapps/entitlements/rest_api/v1/tests/test_views.py

```diff
@@ -672,6 +672,34 @@ def test_get_expired_entitlement_by_uuid(self):
         results = response.data
         assert results.get('expired_at')
 
+    def test_entitlements_filter_on_course_uuid(self):
+        """
+        Verify that courses filtered properly on list of course_uuids.
+        """
+        entitlements = CourseEntitlementFactory.create_batch(5, user=self.user)
+        course_uuids_to_filter_on = {
+            str(entitlement.course_uuid)
+            for entitlement in entitlements[0:3]
+        }
+        # Create a 2nd entitlement for one of the ones to filter on
+        CourseEntitlementFactory.create(user=self.user, course_uuid=str(entitlements[0].course_uuid))
+
+        url = reverse('entitlements_api:v1:entitlements-list')
+        url += f'?user={self.user.username}'
+        url += f'&course_uuid={",".join(course_uuids_to_filter_on)}'
+
+        response = self.client.get(
+            url,
+            content_type='application/json',
+        )
+        assert response.status_code == 200
+
+        results = response.data
+        assert results['count'] == 4
+        actual_course_uuids = {entitlement['course_uuid'] for entitlement in results['results']}
+        assert actual_course_uuids == course_uuids_to_filter_on
+        assert len(actual_course_uuids) == 3
```

Comment:
Should not the length be 4?

Reply:
nope, this is a set, so even though there are 4 courses, 1 is a duplicate, so we don't expect 3 unique UUIDs (though there are 4 results)